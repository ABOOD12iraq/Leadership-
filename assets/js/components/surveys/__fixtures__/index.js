export const invalidQuestionTypeSurvey = {
	survey_payload: {
		completion: [
			{
				completion_ordinal: 1,
				completion_title: 'Thanks for sharing your thoughts!',
				completion_text: 'We’re glad Site Kit is helpful for you! To help others discover it too, take a moment to share your opinion as a review.',
				follow_up_text: 'Let’s go',
				follow_up_url: '#new-url',
				trigger_condition: [
					{
						question_ordinal: 1,
						answer_ordinal: [ 1, 2, 3, 4, 5 ],
					},
				],
			},
		],
		question: [
			{
				question_ordinal: 1,
				question_text: 'Based on your experience so far, how satisfied are you with Site Kit?',
				question_type: 'unknown',
				question: {
					answer_choice: [
						{
							answer_ordinal: 1,
							text: 'Unhappy',
						},
						{
							answer_ordinal: 5,
							text: 'Delighted',
						},
					],
				},
			},
		],
	},
	session: {
		session_id: 'storybook_session',
		session_token: 'token_12345',
	},
};

export const multiQuestionSurvey = {
	survey_payload: {
		completion: [
			{
				completion_ordinal: 1,
				completion_title: 'You answered early!',
				completion_text: 'Because you picked answer 4 or 5 we ended this survey early. Cool!',
				follow_up_text: 'Let’s go',
				follow_up_url: '#new-url',
				trigger_condition: [
					{
						question_ordinal: 1,
						answer_ordinal: [ 4, 5 ],
					},
				],
			},
			{
				completion_ordinal: 2,
				completion_title: 'You answered both questions!',
				completion_text: 'What a completionist!.',
				follow_up_text: 'Get help',
				follow_up_url: '#new-url-2',
				trigger_condition: [
					{
						question_ordinal: 2,
						answer_ordinal: [ 1, 2, 3, 4, 5 ],
					},
				],
			},
		],
		question: [
			{
				question_ordinal: 1,
				question_text: 'If you are really happy, this survey will end early!',
				question_type: 'rating',
				question: {
					answer_choice: [
						{
							answer_ordinal: 1,
							text: 'Unhappy',
						},
						{
							answer_ordinal: 2,
							text: 'Dissatisfied',
						},
						{
							answer_ordinal: 3,
							text: 'Neutral',
						},
						{
							answer_ordinal: 4,
							text: 'Satisfied',
						},
						{
							answer_ordinal: 5,
							text: 'Delighted',
						},
					],
				},
			},
			{
				question_ordinal: 2,
				question_text: 'Another question: how do you feel when it rains?',
				question_type: 'rating',
				question: {
					answer_choice: [
						{
							answer_ordinal: 1,
							text: 'Unhappy',
						},
						{
							answer_ordinal: 2,
							text: 'Dissatisfied',
						},
						{
							answer_ordinal: 3,
							text: 'Neutral',
						},
						{
							answer_ordinal: 4,
							text: 'Satisfied',
						},
						{
							answer_ordinal: 5,
							text: 'Delighted',
						},
					],
				},
			},
		],
	},
	session: {
		session_id: 'storybook_session',
		session_token: 'token_12345',
	},
};

export const singleQuestionSurvey = {
	survey_payload: {
		completion: [
			{
				completion_ordinal: 1,
				completion_title: 'Thanks for sharing your thoughts!',
				completion_text: 'We’re glad Site Kit is helpful for you! To help others discover it too, take a moment to share your opinion as a review.',
				follow_up_text: 'Let’s go',
				follow_up_url: '#new-url',
				trigger_condition: [
					{
						question_ordinal: 1,
						answer_ordinal: [ 4, 5 ],
					},
				],
			},
			{
				completion_ordinal: 2,
				completion_title: 'Thanks for sharing your thoughts!',
				completion_text: 'Do you need help with anything? We’re happy to answer your questions in the forum.',
				follow_up_text: 'Get help',
				follow_up_url: '#new-url-2',
				trigger_condition: [
					{
						question_ordinal: 1,
						answer_ordinal: [ 1, 2, 3 ],
					},
				],
			},
		],
		question: [
			{
				question_ordinal: 1,
				question_text: 'Based on your experience so far, how satisfied are you with Site Kit?',
				question_type: 'rating',
				question: {
					answer_choice: [
						{
							answer_ordinal: 1,
							text: 'Unhappy',
						},
						{
							answer_ordinal: 2,
							text: 'Dissatisfied',
						},
						{
							answer_ordinal: 3,
							text: 'Neutral',
						},
						{
							answer_ordinal: 4,
							text: 'Satisfied',
						},
						{
							answer_ordinal: 5,
							text: 'Delighted',
						},
					],
				},
			},
		],
	},
	session: {
		session_id: 'storybook_session',
		session_token: 'token_12345',
	},
};

export const singleQuestionSurveyWithNoFollowUp = {
	survey_payload: {
		completion: [
			{
				completion_ordinal: 1,
				completion_title: 'Thanks for the ranking!',
				completion_text: 'No further questions; this message will now self-destruct.',
				trigger_condition: [
					{
						question_ordinal: 1,
						answer_ordinal: [ 1, 2, 3, 4, 5 ],
					},
				],
			},
		],
		question: [
			{
				question_ordinal: 1,
				question_text: 'Based on your experience so far, how satisfied are you with Site Kit?',
				question_type: 'rating',
				question: {
					answer_choice: [
						{
							answer_ordinal: 1,
							text: 'Unhappy',
						},
						{
							answer_ordinal: 2,
							text: 'Dissatisfied',
						},
						{
							answer_ordinal: 3,
							text: 'Neutral',
						},
						{
							answer_ordinal: 4,
							text: 'Satisfied',
						},
						{
							answer_ordinal: 5,
							text: 'Delighted',
						},
					],
				},
			},
		],
	},
	session: {
		session_id: 'storybook_session',
		session_token: 'token_12345',
	},
};

