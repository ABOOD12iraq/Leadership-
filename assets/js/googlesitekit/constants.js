/**
 * Core constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const SCREEN_CONTEXT_ACTIVATION = 'activation';
export const SCREEN_CONTEXT_ADMIN_BAR = 'adminBar';
export const SCREEN_CONTEXT_DASHBOARD = 'dashboard';
export const SCREEN_CONTEXT_PAGE_DASHBOARD = 'pageDashboard';
export const SCREEN_CONTEXT_MODULE_DASHBOARD = 'moduleDashboard';
export const SCREEN_CONTEXT_SETTINGS = 'settings';
export const SCREEN_CONTEXT_SPLASH = 'splash';
export const SCREEN_CONTEXT_USER_INPUT = 'userInput';
export const SCREEN_CONTEXT_WP_DASHBOARD = 'wpDashboard';
