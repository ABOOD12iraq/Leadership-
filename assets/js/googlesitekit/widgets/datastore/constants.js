/**
 * `core/widgets` data store: constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const WIDGET_AREA_STYLES = { BOXES: 'boxes', COMPOSITE: 'composite' };
export const WIDGET_WIDTHS = { QUARTER: 'quarter', HALF: 'half', FULL: 'full' };

export const CORE_WIDGETS = 'core/widgets';

// Key Metrics Widgets
export const KM_ANALYTICS_ENGAGED_TRAFFIC_SOURCE =
	'kmAnalyticsEngagedTrafficSource';
export const KM_ANALYTICS_LOYAL_VISITORS = 'kmAnalyticsLoyalVisitors';
export const KM_ANALYTICS_NEW_VISITORS = 'kmAnalyticsNewVisitors';
export const KM_ANALYTICS_POPULAR_CONTENT = 'kmAnalyticsPopularContent';
export const KM_ANALYTICS_POPULAR_PRODUCTS = 'kmAnalyticsPopularProducts';
export const KM_ANALYTICS_TOP_CITIES = 'kmAnalyticsTopCities';
export const KM_ANALYTICS_TOP_CONVERTING_TRAFFIC_SOURCE =
	'kmTopConvertingTrafficSource';
export const KM_ANALYTICS_TOP_COUNTRIES = 'kmAnalyticsTopCountries';
export const KM_ANALYTICS_TOP_TRAFFIC_SOURCE = 'kmAnalyticsTopTrafficSource';
export const KM_SEARCH_CONSOLE_POPULAR_KEYWORDS =
	'kmSearchConsolePopularKeywords';
