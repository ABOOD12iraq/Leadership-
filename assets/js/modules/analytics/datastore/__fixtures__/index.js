/**
 * Analytics Datastore Fixtures.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export { default as accountsPropertiesProfiles } from './accounts-properties-profiles';
export { default as adminBarAnalyticsTotalUsersMockData } from './admin-bar-analytics-total-users-mock-data';
export { default as adminBarAnalyticsTotalUsersOptions } from './admin-bar-analytics-total-users-options';
export { default as adminBarAnalyticsSessionsMockData } from './admin-bar-analytics-session-mock-data';
export { default as adminBarAnalyticsSessionsOptions } from './admin-bar-analytics-session-options';
export { default as createProfile } from './create-profile';
export { default as createProperty } from './create-property';
export { default as goals } from './goals';
export { default as profiles } from './profiles';
export { default as propertiesProfiles } from './properties-profiles';
export { default as getTagPermissionsAccess } from './tag-permissions-access';
export { default as getTagPermissionsNoAccess } from './tag-permissions-no-access';
export { default as createAccount } from './create-account';
export { default as report } from './report';
export { default as dashboardAllTrafficArgs } from './dashboard-all-traffic-widget-args.json';
export { default as dashboardAllTrafficData } from './dashboard-all-traffic-widget-data.json';
export { default as pageDashboardAllTrafficArgs } from './page-dashboard-all-traffic-widget-args.json';
export { default as pageDashboardAllTrafficData } from './page-dashboard-all-traffic-widget-data.json';
export { default as dashboardPopularPagesArgs } from './dashboard-popular-pages-widget-args.json';
export { default as dashboardPopularPagesData } from './dashboard-popular-pages-widget-data.json';
export { default as pageDashboardBounceRateWidgetArgs } from './page-dashboard-bounce-rate-widget-args.json';
export { default as pageDashboardBounceRateWidgetData } from './page-dashboard-bounce-rate-widget-data.json';
export { default as dashboardBounceRateWidgetData } from './dashboard-bounce-rate-widget-data.json';
export { default as dashboardBounceRateWidgetArgs } from './dashboard-bounce-rate-widget-args.json';
export { default as dashboardGoalsWidgetData } from './dashboard-goals-widget-data.json';
export { default as dashboardGoalsWidgetArgs } from './dashboard-goals-widget-args.json';
export { default as dashboardUserDimensionsData } from './dashboard-user-dimensions-data.json';
export { default as dashboardUserDimensionsArgs } from './dashboard-user-dimensions-args.json';
export { default as dashboardUserTotalsData } from './dashboard-user-totals-data.json';
export { default as dashboardUserTotalsArgs } from './dashboard-user-totals-args.json';
export { default as dashboardUserGraphData } from './dashboard-user-graph-data.json';
export { default as dashboardUserGraphArgs } from './dashboard-user-graph-args.json';
export { default as wpDashboardPopularPagesArgs } from './wp-dashboard-popular-pages-args.json';
export { default as wpDashboardPopularPagesData } from './wp-dashboard-popular-pages-data.json';
export { default as wpDashboardUniqueVisitorsArgs } from './wp-dashboard-unique-visitors-args.json';
export { default as wpDashboardUniqueVisitorsData } from './wp-dashboard-unique-visitors-data.json';
export { default as wpDashboardSessionDurationArgs } from './wp-dashboard-session-duration-args.json';
export { default as wpDashboardSessionDurationData } from './wp-dashboard-session-duration-data.json';
export * from './dashboard-unique-visitors-data';
export * from './dashboard-unique-visitors-args';
export { pageDashboardUniqueVisitorsSparkData, pageDashboardUniqueVisitorsVisitorData } from './page-dashboard-unique-visitors-data';
export { pageDashboardUniqueVisitorsSparkArgs, pageDashboardUniqueVisitorsVisitorArgs } from './page-dashboard-unique-visitors-args';
import defaultSettings from './settings--default.json';

export const settings = {
	default: defaultSettings,
};
