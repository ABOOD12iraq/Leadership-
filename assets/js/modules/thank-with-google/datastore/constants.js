/**
 * `modules/thank-with-google` data store constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_THANK_WITH_GOOGLE = 'modules/thank-with-google';

export const STATE_ACTIVE = 'ACTIVE';
export const STATE_ACTION_REQUIRED = 'ACTION_REQUIRED';
export const STATE_PENDING_VERIFICATION = 'PENDING_VERIFICATION';

export const BUTTON_PLACEMENT_DYNAMIC_LOW = 'dynamic_low';
export const BUTTON_PLACEMENT_DYNAMIC_HIGH = 'dynamic_high';
export const BUTTON_PLACEMENT_STATIC_AUTO = 'static_auto';
export const BUTTON_PLACEMENT_STATIC_ABOVE_CONTENT = 'static_above-content';
export const BUTTON_PLACEMENT_STATIC_BELOW_CONTENT = 'static_below-content';
export const BUTTON_PLACEMENT_STATIC_BELOW_1ST_PARAGRAPH =
	'static_below-first-paragraph';
