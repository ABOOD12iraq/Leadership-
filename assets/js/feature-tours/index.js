/**
 * Feature Tours.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { VIEW_CONTEXT_DASHBOARD } from '../googlesitekit/constants';
import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';

const allTrafficWidget = {
	slug: 'allTrafficWidget',
	contexts: [ VIEW_CONTEXT_DASHBOARD ],
	version: '1.25.0',
	checkRequirements: async ( registry ) => {
		// Here we need to wait for the underlying selector to be resolved before selecting `isModuleConnected`.
		await registry.__experimentalResolveSelect( CORE_MODULES ).getModules();
		if ( ! registry.select( CORE_MODULES ).isModuleConnected( 'analytics' ) ) {
			return false;
		}
		return true;
	},
	steps: [
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__user-count-chart',
			title: __( "It's now easier to see your site's traffic at a glance", 'google-site-kit' ),
			content: __( 'Check the trend graph to see how your traffic changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__dimensions-chart',
			title: __( 'See where your visitors come from', 'google-site-kit' ),
			content: __( 'Click on the chart slices to see how each segment has changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-header__date-range-selector-menu',
			title: __( 'Check how your traffic changed since you last looked', 'google-site-kit' ),
			content: __( 'Select a time frame to see the comparison with the previous time period', 'google-site-kit' ),
		},
	],
};

// Ordered tours.
export default [
	allTrafficWidget,
];
