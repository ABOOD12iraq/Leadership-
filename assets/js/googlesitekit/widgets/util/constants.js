/**
 * Widgets layout constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { WIDGET_WIDTHS } from '../datastore/constants';

export const WIDTH_GRID_COUNTER_MAP = {
	[ WIDGET_WIDTHS.QUARTER ]: 3,
	[ WIDGET_WIDTHS.HALF ]: 6,
	[ WIDGET_WIDTHS.FULL ]: 12,
};

export const WIDTH_GRID_CLASS_MAP = {
	[ WIDGET_WIDTHS.QUARTER ]: [
		'mdc-layout-grid__cell',
		'mdc-layout-grid__cell--span-2-phone',
		'mdc-layout-grid__cell--span-3-desktop',
		'mdc-layout-grid__cell--span-4-tablet',
	],
	[ WIDGET_WIDTHS.HALF ]: [
		'mdc-layout-grid__cell',
		'mdc-layout-grid__cell--span-6-desktop',
		'mdc-layout-grid__cell--span-8-tablet',
	],
	[ WIDGET_WIDTHS.FULL ]: [
		'mdc-layout-grid__cell',
		'mdc-layout-grid__cell--span-12',
	],
};

export const HIDDEN_CLASS = 'googlesitekit-hidden';
