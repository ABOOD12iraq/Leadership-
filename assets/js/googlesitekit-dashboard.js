/**
 * Dashboard component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { clearWebStorage } from './util';
import Root from './components/Root';
import './components/legacy-notifications';
import {
	VIEW_CONTEXT_DASHBOARD,
	VIEW_CONTEXT_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_MODULE_SETUP,
} from './googlesitekit/constants';
import DashboardEntryPoint from './components/DashboardEntryPoint';

// Initialize the app once the DOM is ready.
domReady( async () => {
	if ( global._googlesitekitLegacyData.admin.resetSession ) {
		clearWebStorage();
	}

	const renderTarget = document.getElementById(
		'js-googlesitekit-dashboard'
	);

	if ( renderTarget ) {
		// Make a separate API request since we cannot use selectors
		// outside the Root component.
		const { authenticated: isAuthenticated } = await API.get(
			'core',
			'user',
			'authentication',
			{},
			{ useCache: false }
		);

		const { setupModuleSlug } = renderTarget.dataset;
		let viewContext = VIEW_CONTEXT_MODULE_SETUP;
		if ( ! setupModuleSlug ) {
			viewContext = isAuthenticated
				? VIEW_CONTEXT_DASHBOARD
				: VIEW_CONTEXT_DASHBOARD_VIEW_ONLY;
		}
		render(
			<Root viewContext={ viewContext }>
				<DashboardEntryPoint setupModuleSlug={ setupModuleSlug } />
			</Root>,
			renderTarget
		);
	}
} );
