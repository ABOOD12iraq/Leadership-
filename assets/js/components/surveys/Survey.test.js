/**
 * Survey component tests.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import Survey from './Survey';
import { render, fireEvent, createTestRegistry } from '../../../../tests/js/test-utils';
import * as fixtures from './__fixtures__';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import { CORE_FORMS } from '../../googlesitekit/datastore/forms/constants';

describe( 'Survey', () => {
	let registry;

	beforeEach( () => {
		registry = createTestRegistry();
	} );

	it( 'should render a survey when one exists in the datastore', async () => {
		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );

		fetchMock.postOnce( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );

		const { container } = render( <Survey />, { registry } );

		expect( fetchMock ).toHaveFetched( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/ );

		expect( container ).toMatchSnapshot();
	} );

	it( "should send a 'survey_shown' event on mount", async () => {
		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );

		fetchMock.postOnce( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );

		render( <Survey />, { registry } );

		expect( fetchMock ).toHaveFetched(
			'/google-site-kit/v1/core/user/data/survey-event?_locale=user',
			{
				body: {
					data: {
						event: { survey_shown: {} },
						session: fixtures.singleQuestionSurvey.session,
					},
				},
				credentials: 'include',
				headers: {
					Accept: 'application/json, */*;q=0.1',
					'Content-Type': 'application/json',
				},
				method: 'POST',
			}
		);
	} );

	it( "should send a 'question_answered' event when a question is answered", async () => {
		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );

		fetchMock.post( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );

		const { getByLabelText } = render( <Survey />, { registry } );

		fireEvent.click( getByLabelText( 'Delighted icon' ) );

		expect( fetchMock ).toHaveFetched(
			'/google-site-kit/v1/core/user/data/survey-event?_locale=user',
			{
				body: {
					data: {
						event: {
							question_answered: {
								question_ordinal: 1,
								answer: {
									answer_ordinal: 5,
								},
							},
						},
						session: fixtures.singleQuestionSurvey.session,
					},
				},
				credentials: 'include',
				headers: {
					Accept: 'application/json, */*;q=0.1',
					'Content-Type': 'application/json',
				},
				method: 'POST',
			}
		);
	} );

	it( 'should mark the question as answered in the core/forms datastore', async () => {
		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );

		fetchMock.post( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );

		const { getByLabelText } = render( <Survey />, { registry } );

		fireEvent.click( getByLabelText( 'Delighted icon' ) );

		expect(
			registry
				.select( CORE_FORMS )
				.getValue(
					`survey-${ fixtures.singleQuestionSurvey.session.session_id }`,
					'answers'
				)
		).toEqual(
			[ { question_ordinal: 1, answer_ordinal: 5 } ]
		);
	} );

	it( 'should render nothing if no survey exists', () => {
		const { container } = render( <Survey />, { registry } );

		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should render nothing if the survey is dismissed', () => {
		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );

		fetchMock.post( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );

		const { container, getByLabelText } = render( <Survey />, { registry } );

		fireEvent.click( getByLabelText( 'Dismiss this survey' ) );

		expect( container ).toBeEmptyDOMElement();
	} );
} );
