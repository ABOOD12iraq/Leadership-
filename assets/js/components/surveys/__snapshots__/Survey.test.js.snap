// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey should render a survey when one exists in the datastore 1`] = `
<div>
  <div
    class="googlesitekit-survey__question-rating"
  >
    <div
      class="googlesitekit-survey__header"
    >
      <div
        class="googlesitekit-survey__header-logo"
      >
        <svg />
      </div>
      <div
        class="googlesitekit-survey__header-details"
      >
        <h3>
          Based on your experience so far, how satisfied are you with Site Kit?
        </h3>
        <button
          aria-label="Dismiss this survey"
          class="mdc-button googlesitekit-survey__header-close mdc-button--raised"
          target="_self"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            height="40"
            role="img"
            viewBox="0 0 24 24"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 11.9l3.3-3.4-1.1-1-3.2 3.3-3.2-3.3-1.1 1 3.3 3.4-3.5 3.6 1 1L12 13l3.5 3.5 1-1z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="googlesitekit-survey__body"
    >
      <div
        class="googlesitekit-survey__choices"
      >
        <div
          class="googlesitekit-survey__choice"
        >
          <button
            aria-label="Unhappy icon"
            class="mdc-button mdc-button--raised"
            target="_self"
          >
            <svg />
          </button>
          <p>
            Unhappy
          </p>
        </div>
        <div
          class="googlesitekit-survey__choice"
        >
          <button
            aria-label="Dissatisfied icon"
            class="mdc-button mdc-button--raised"
            target="_self"
          >
            <svg />
          </button>
          <p>
            Dissatisfied
          </p>
        </div>
        <div
          class="googlesitekit-survey__choice"
        >
          <button
            aria-label="Neutral icon"
            class="mdc-button mdc-button--raised"
            target="_self"
          >
            <svg />
          </button>
          <p>
            Neutral
          </p>
        </div>
        <div
          class="googlesitekit-survey__choice"
        >
          <button
            aria-label="Satisfied icon"
            class="mdc-button mdc-button--raised"
            target="_self"
          >
            <svg />
          </button>
          <p>
            Satisfied
          </p>
        </div>
        <div
          class="googlesitekit-survey__choice"
        >
          <button
            aria-label="Delighted icon"
            class="mdc-button mdc-button--raised"
            target="_self"
          >
            <svg />
          </button>
          <p>
            Delighted
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
